src/include/reconcile_inline.h

```diff
@@ -239,14 +239,14 @@ __wt_rec_incr(WT_SESSION_IMPL *session, WT_RECONCILE *r, uint32_t v, size_t size
 }
 
 /*
- * __wt_rec_image_copy --
- *     Copy a key/value cell and buffer pair into the new image.
+ * __wt_rec_kv_copy --
+ *     Copy a key/value cell and buffer pair.
  */
 static WT_INLINE void
-__wt_rec_image_copy(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_KV *kv)
+__wt_rec_kv_copy(WT_SESSION_IMPL *session, uint8_t *p, WT_REC_KV *kv)
```

Comment:
What guarantees that `p` points to enough memory to hold the key/value? This feels easy to misuse but I don't know much about the reconciliation code.

Reply:
That's a good point. But I don't know how to improve this.

Reply:
Yeah, from my experience of working with these kinds of things, I don't think we make any guarantees that it will hold enough memory.

Reply:
What if we did something like `uint8_t *p[MAX_SIZE_EXPECTED];` and then passed that in?

Reply:
max size expected is not a fixed number.

Reply:
If we go over the boundary, we should crash.

Reply:
This is a different thing on the higher level. Generally, the current code ensures the buffer is enough in the higher level of the stack.

Reply:
Ah, OK. I guess my inexperience with the reconciliation code is showing :-)

I guess it's no worse than the previous version which also assumed enough space, but at least had a typed structure instead of some raw buffer.

Longer-term I think this needs attention but I'm OK with leaving it for the PR - if you agree, can you add a `TODO` comment please?